Jenkins_setup.md

```diff
@@ -0,0 +1,161 @@
+plugin.
+
+The GHPRB plugin is configured to be triggered via github hooks. This is the preferred method
+to trigger on changes (preferred to polling). The preferred method is to integrate the CIs
+using the [katacontainersbot](https://github.com/katacontainersbot) user. Please discuss
+your requirements with the Kata Containers team who can then discuss the appropriate changes.
+
+The GHPRB triggers should be set up similar to:
+![Jenkins GHPRB plugin trigger config](pictures/ghprb_trigger.png)
+
+The GHPRB `Trigger Setup` further options should be set up similar to:
+![Jenkins GHPRB plugin trigger setup config](pictures/ghprb_adv_triggers.png)
+
+### Master builds
+
+Master branch builds are tracked with the Jenkins [git Source Code Management](https://plugins.jenkins.io/git)
+plugin. GITScm GitHub hooks are used to trigger builds.
+
+The Jenkins setup should look similar to the following for the:
+
+- Repository URL setup
+- Branches to build
+- Hook trigger enablement
+
+![Jenkins master setup](pictures/master_git_config.png)
+
+
+### Github bandwidth and tokens
+
+It should be noted that the GitHub API access is [rate limited](https://developer.github.com/v3/#rate-limiting). If no github access authentication method is configured into Jenkins
+then the CI may quickly run out of GitHub access bandwidth, which will stall the CI.
+
+For Kata Containers CI we configure Jenkins with a GitHub [Personal API token](https://blog.github.com/2013-05-16-personal-api-tokens/), which increases the bandwidth capacity.
+
+## Job build script
+
+Kata Containers Jenkins builds, for both PR and Master builds, are invoked using the top
+level [`jenkins_job_build.sh`](https://github.com/kata-containers/tests/blob/master/.ci/jenkins_job_build.sh) script from the [katacontainers/tests](https://github.com/kata-containers/tests)
+repository.
+
+This script takes a single argument (the path to the repository to build), and utilises
+some environment variables to guide what sort of build and run it should perform.
+
+An example of how this script is invoked from Jenkins is shown in the [Setting up Jenkins](#setting-up-jenkins) section.
+
+## Setting up Jenkins
+
+### Naming jobs
+
+Generally the Jobs in Jenkins are named in the form `kata-containers-<repo>-<distro>-<buildtype>`,
+e.g., for a runtime repo PR build on Ubuntu 17.10, the Job would be called:
+`kata-containers-runtime-ubuntu-17-10-PR`. Following this naming convention will make it much
+easier for other members of the project to navigate all CIs in a similar manner.
+
+### Setting up the scripts
+
+Both PR and Master builds use the same form of build and post-build teardown scripts that
+are configured in the Jenkins Job UI. These scripts look like:
+
+Build execute shell:
+```bash
+#!/bin/bash
+
+set -e
+
+export ghprbPullId
+export ghprbTargetBranch
+
+cd $HOME
+git clone https://github.com/kata-containers/tests.git
```

Comment:
And I agree! - but, the above is how it is done today in our CI (in the QA CI anyhow - iirc, in the metrics CI I use a 'go get' if I can rather than a raw git clone).
/cc @chavafg @sboeuf for input. I will leave as is at present in this doc unless we change it in our live Jenkins.